packages/server-core/src/setting/authentication-setting/authentication-setting.class.ts

```diff
@@ -23,71 +23,72 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
 Ethereal Engine. All Rights Reserved.
 */
 
-import { Paginated, Params } from '@feathersjs/feathers'
+import type { Id, Params } from '@feathersjs/feathers'
+import { KnexAdapter } from '@feathersjs/knex'
+import type { KnexAdapterOptions, KnexAdapterParams } from '@feathersjs/knex'
 import * as k8s from '@kubernetes/client-node'
-import { SequelizeServiceOptions, Service } from 'feathers-sequelize'
 
-import { AdminAuthSetting as AdminAuthSettingInterface } from '@etherealengine/common/src/interfaces/AdminAuthSetting'
 import { UserInterface } from '@etherealengine/common/src/interfaces/User'
+import {
+  AuthenticationSettingData,
+  AuthenticationSettingPatch,
+  AuthenticationSettingQuery,
+  AuthenticationSettingType
+} from '@etherealengine/engine/src/schemas/setting/authentication-setting.schema'
 import { getState } from '@etherealengine/hyperflux'
 
 import { Application } from '../../../declarations'
 import config from '../../appconfig'
 import logger from '../../ServerLogger'
 import { ServerState } from '../../ServerState'
-import { UserParams } from '../../user/user/user.class'
 
-export type AdminAuthSettingDataType = AdminAuthSettingInterface
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface AuthenticationSettingParams extends KnexAdapterParams<AuthenticationSettingQuery> {
+  user: UserInterface
+}
 
-export class Authentication<T = AdminAuthSettingDataType> extends Service<T> {
+/**
+ * A class for AuthenticationSetting service
+ */
+
+export class AuthenticationSettingService<
+  T = AuthenticationSettingType,
+  ServiceParams extends Params = AuthenticationSettingParams
+> extends KnexAdapter<
+  AuthenticationSettingType,
+  AuthenticationSettingData,
+  AuthenticationSettingParams,
+  AuthenticationSettingPatch
+> {
   app: Application
 
-  constructor(options: Partial<SequelizeServiceOptions>, app: Application) {
+  constructor(options: KnexAdapterOptions, app: Application) {
     super(options)
     this.app = app
   }
 
-  async find(params?: UserParams): Promise<T[] | Paginated<T>> {
-    const auth = (await super.find()) as any
-    const loggedInUser = params!.user as UserInterface
+  async find(params?: AuthenticationSettingParams) {
```

Comment:
I'm encountering issues with authentication-setting.find with non-admin users. It's returning `[{"message":"Cannot convert undefined or null to object"}]`. It might be because calls with the admin scope are returning data that calls without it do not. Could it be that certain fields are defined as required that shouldn't be?

Reply:
I pushed a fixed. Kindly review.